setup.py

```diff
@@ -53,66 +54,85 @@ def run(self):
                 try:            os.remove(g)
                 except OSError: pass
 
+# Platform information
+python2 = (sys.version_info[0] == 2)
+windows = sys.platform.startswith("win")
 
 # If cython is present, we'll compile
 # the pyx files from scratch. Otherwise,
 # we'll compile the pre-generated c
 # files (which are assumed to be present).
-
-# We only need numpy to
-# compile the test modules
 have_cython = True
-have_numpy  = True
-
+build_tests  = True
 
 try:    from Cython.Build import cythonize
 except: have_cython = False
 
+# We need numpy to compile the test modules
 try:    import numpy as np
-except: have_numpy = False
+except: build_tests = False
 
+include_dirs = ['indexed_gzip']
+lib_dirs = []
+libs = []
+extra_compile_args = []
 
 # If numpy is present, we need
 # to include the headers
-include_dirs = ['indexed_gzip']
-if have_numpy:
+if build_tests:
     include_dirs.append(np.get_include())
 
+if windows:
+    build_tests = False
+    ZLIB_HOME = os.environ.get("FSLDIR", "c:/Program Files (x86)/GnuWin32")
+    include_dirs.append(os.path.join(ZLIB_HOME, "include"))
+    libs.append('zlib')
+    lib_dirs.append(os.path.join(ZLIB_HOME, "lib"))
+    if python2:
+        # For stdint.h which is not included in the old Visual C 
+        # compiler used for Python 2
+        include_dirs.append('compat')
+        extra_compile_args += ['-DNO_C99']
+else:
+    libs.append('z')
+    extra_compile_args += ['-Wall', '-pedantic', '-Wno-unused-function']
 
 # Compile from cython files if
 # possible, or compile from c.
 if have_cython: pyx_ext = 'pyx'
 else:           pyx_ext = 'c'
 
-
 # The indexed_gzip module
 igzip_ext = Extension(
     'indexed_gzip.indexed_gzip',
     [op.join('indexed_gzip', 'indexed_gzip.{}'.format(pyx_ext)),
      op.join('indexed_gzip', 'zran.c')],
-    libraries=['z'],
-    extra_compile_args=['-Wall', '-pedantic', '-Wno-unused-function'])
+    libraries=libs,
+    library_dirs=lib_dirs,
+    include_dirs=include_dirs,
+    extra_compile_args=extra_compile_args)
 
 # Optional test modules
 test_exts = [
     Extension(
         'indexed_gzip.tests.ctest_zran',
         [op.join('indexed_gzip', 'tests', 'ctest_zran.{}'.format(pyx_ext)),
          op.join('indexed_gzip', 'zran.c')],
-        libraries=['z'],
+        libraries=libs,
+        library_dirs=lib_dirs,
         include_dirs=include_dirs,
-        extra_compile_args=['-Wno-unused-function']),
+        extra_compile_args=extra_compile_args),
     Extension(
```

Comment:
Would you be able to try compiling/running the `ctest_indexed_gzip` tests? This one should hopefully be able to be compiled/run on windows.